### PR TITLE
[TypeScript] Fix remaining strictNullCheck errors in ra-ui-materialui

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-ui-materialui/src/field/ImageField.tsx [hunk 1]
```diff
@@ -48,8 +49,10 @@ export const ImageField = <
             <Root className={className} {...sanitizeFieldRestProps(rest)}>
                 <ul className={ImageFieldClasses.list}>
                     {sourceValue.map((file, index) => {
-                        const fileTitleValue = get(file, title) || title;
-                        const srcValue = get(file, src) || title;
+                        const fileTitleValue = title
+                            ? get(file, title, title)
+                            : title;
+                        const srcValue = src ? get(file, src, src) : src;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/ImageField.tsx:55]: you've changed the meaning there, shouldn't this be 
```suggestion
                        const srcValue = src ? get(file, src, title) : title;
```

File: packages/ra-ui-materialui/src/field/TranslatableFields.tsx [hunk 2]
```diff
@@ -78,7 +78,17 @@ export const TranslatableFields = (
         className,
     } = props;
     const record = useRecordContext(props);
+    if (!record) {
+        throw new Error(
+            `<TranslatableFields> was called outside of a ResourceContext and without a resource prop. You must set the resource prop.`
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/TranslatableFields.tsx:83]:
```suggestion
            `<TranslatableFields> was called outside of a RecordContext and without a record prop. You must set the record prop.`
```

File: packages/ra-ui-materialui/src/field/TranslatableFields.tsx [hunk 3]
```diff
@@ -78,7 +78,17 @@ export const TranslatableFields = (
...
+        );
+    }
     const resource = useResourceContext(props);
+    if (!resource) {
+        throw new Error(
+            `<TranslatableFields> was called outside of a ResourceContext and without a resource prop. You must set the resource prop.`
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/field/TranslatableFields.tsx:89]:
```suggestion
            `<TranslatableFields> was called outside of a RecordContext and without a record prop. You must set the record prop.`
```

File: packages/ra-core/src/preferences/PreferencesEditorContext.tsx [hunk 4]
```diff
@@ -2,22 +2,20 @@ import * as React from 'react';
...
-          enable: () => void;
-          disable: () => void;
-          path: string | null;
-          setPath: (path: string) => void;
-      }
-    | undefined;
+export type PreferencesEditorContextValue = {
+    editor: ReactElement | null;
+    setEditor: React.Dispatch<React.SetStateAction<ReactElement | null>>;
+    preferenceKey: string | null;
```
Comment (fzaninotto) [inline packages/ra-core/src/preferences/PreferencesEditorContext.tsx:11]: I wonder why you prefer null to undefined here? Is this change necessary?

File: packages/ra-core/src/preferences/PreferencesEditorContextProvider.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import {
 export const PreferencesEditorContextProvider = ({ children }) => {
     const [isEnabled, setIsEnabled] = useState(false);
     const [editor, setEditor] = useState<ReactElement | null>(null);
-    const [preferenceKey, setPreferenceKey] = useState<string>();
+    const [preferenceKey, setPreferenceKey] = useState<string | null>(null);
```
Comment (fzaninotto) [inline packages/ra-core/src/preferences/PreferencesEditorContextProvider.tsx:11]: you could default to undefined

File: packages/ra-ui-materialui/src/form/SimpleFormConfigurable.tsx [hunk 6]
```diff
@@ -31,27 +31,28 @@ export const SimpleFormConfigurable = ({
...
+                                  : translate(
+                                        'ra.configurable.SimpleForm.unlabeled',
+                                        {
+                                            input: index,
+                                            _: `Unlabeled input #%{input}`,
+                                        }
+                                    ),
+                      }
+                    : null
+            )?.filter(column => column != null) ?? [];
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleFormConfigurable.tsx:52]: I think you should use an emptyArray const defined outside of the element to avoid rerenderings when the effect returns a different empty array object (modified 3 times)

File: packages/ra-ui-materialui/src/preferences/Configurable.tsx [hunk 7]
```diff
@@ -73,6 +73,7 @@ export const Configurable = (props: ConfigurableProps) => {
     }
 
     const handleOpenEditor = () => {
+        if (!setEditor) return;
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/preferences/Configurable.tsx:76]: I think we should rather throw an error in this case

File: packages/ra-ui-materialui/src/detail/CreateView.tsx [hunk 8]
```diff
@@ -22,10 +22,7 @@ export const CreateView = (props: CreateViewProps) => {
     const { resource, defaultTitle } = useCreateContext();
 
     return (
-        <Root
-            className={clsx('create-page', className)}
-            {...sanitizeRestProps(rest)}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/detail/CreateView.tsx:27]: I'm surprised this works, since CreateViewProps is declared as CreateProps (which is a mistake, as there are some Create props that CreateView will never get injected)

**Review by fzaninotto — CHANGES_REQUESTED** [review]
We're getting close!

File: packages/ra-ui-materialui/src/detail/CreateView.tsx (hunk 8 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/detail/CreateView.tsx:27]: And yet it does without any warning at runtime in all demos

File: packages/ra-core/src/preferences/PreferencesEditorContext.tsx (hunk 4 above)
Comment (djhi) [reply packages/ra-core/src/preferences/PreferencesEditorContext.tsx:11]: We could but it looked more coherent with the other props

File: packages/ra-ui-materialui/src/field/FileField.tsx [hunk 9]
```diff
@@ -67,8 +68,10 @@ export const FileField = <
         return (
             <StyledList className={className} {...sanitizeFieldRestProps(rest)}>
                 {sourceValue.map((file, index) => {
-                    const fileTitleValue = get(file, title) || title;
-                    const srcValue = get(file, src) || title;
+                    const fileTitleValue = title
+                        ? get(file, title, title)
+                        : title;
+                    const srcValue = src ? get(file, src, src) : src;
```
Comment (djhi) [inline packages/ra-ui-materialui/src/field/FileField.tsx:74]:
```suggestion
                    const srcValue = src ? get(file, src, title) : title;
```

File: packages/ra-ui-materialui/src/preferences/Configurable.tsx [hunk 10]
```diff
@@ -73,6 +73,12 @@ export const Configurable = (props: ConfigurableProps) => {
     }
 
     const handleOpenEditor = () => {
+        if (!setEditor) {
+            throw new Error(
+                'PreferencesEditorContext is not available. Did you forget to put a PreferencesEditorContextProvider at a higher level?'
```
Comment (adguernier) [inline packages/ra-ui-materialui/src/preferences/Configurable.tsx:78]: **suggestion**(niptick): I like the way you present this error but other errors of this type throw this message: `Configurable must be used inside a PreferencesEditorContextProvider`. Could use the same shape to stay consistent?

```suggestion
                'Configurable must be used inside a PreferencesEditorContextProvider'
```

File: packages/ra-ui-materialui/src/preferences/Configurable.tsx [hunk 11]
```diff
@@ -83,6 +89,11 @@ export const Configurable = (props: ConfigurableProps) => {
                 key: prefixedPreferenceKey,
             })
         );
+        if (!setPreferenceKey) {
+            throw new Error(
+                'PreferencesEditorContext is not available. Did you forget to put a PreferencesEditorContextProvider at a higher level?'
```
Comment (adguernier) [inline packages/ra-ui-materialui/src/preferences/Configurable.tsx:94]: (niptick)
```suggestion
                'Configurable must be used inside a PreferencesEditorContextProvider'
```

**Review by adguernier — APPROVED** [review]

File: packages/ra-ui-materialui/src/detail/CreateView.tsx [hunk 12]
```diff
@@ -41,8 +40,15 @@ export const CreateView = (props: CreateViewProps) => {
     );
 };
 
-export type CreateViewProps = CreateProps;
-
+export interface CreateViewProps {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/detail/CreateView.tsx:43]: I think it should extend the base div props (e.g. `style`), otherwise valid HTML attributes, that are passed down to the root components, con't be accepted. This also allows you to get rid of `className`. Same for the other 2.

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
\o/
